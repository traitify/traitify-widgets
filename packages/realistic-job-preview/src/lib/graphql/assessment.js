export const create = `
  mutation createAssessment(
    $localeKey: String!,
    $profileID: String!,
    $surveyID: String!
  ) {
    createAssessment(
      localeKey: $localeKey,
      profileId: $profileID,
      surveyId: $surveyID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;

export const get = `
  query assessment(
    $assessmentID: ID!
  ) {
    assessment(
      id: $assessmentID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;

export const path = "/realistic-job-previews/graphql";

export const start = `
  mutation startAssessment(
    $assessmentID: ID!
  ) {
    startAssessment(
      id: $assessmentID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;

export const update = `
  mutation updateAssessmentAnswer(
    $answers: [AssessmentAnswer],
    $assessmentID: String!
  ) {
    updateAssessmentAnswer(
      answers: $answers,
      assessmentId: $assessmentID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;
