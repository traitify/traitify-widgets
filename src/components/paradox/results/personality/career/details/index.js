import PropTypes from "prop-types";
import {Component} from "react";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

class Career extends Component {
  static propTypes = {
    career: PropTypes.shape({
      description: PropTypes.string.isRequired,
      experience_level: PropTypes.shape({
        id: PropTypes.number.isRequired
      }).isRequired,
      picture: PropTypes.string.isRequired,
      score: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired
    }).isRequired,
    translate: PropTypes.func.isRequired,
    ui: TraitifyPropTypes.ui.isRequired,
    setElement: PropTypes.func.isRequired
  };
  componentDidMount() {
    this.props.ui.trigger("Career.initialized", this);
  }
  componentDidUpdate() {
    this.props.ui.trigger("Career.updated", this);
  }
  openModal = () => {
    this.props.ui.trigger("CareerModal.career", this, this.props.career);
    this.props.ui.trigger("CareerModal.show", this);
  };
  render() {
    const {career, translate, setElement} = this.props;

    return (
      <div className={style.container} id={style.container} ref={setElement}>
        <div className={style.careerContainer}>
          <img alt={career.title} src={career.picture} />
          <div className={style.careerDetails}>
            <div className={style.title}>{career.title}</div>
            <div className={style.description}>{career.description}</div>
          </div>
        </div>
        <hr className={style.grayDivider} />
        <div className={style.content}>
          <div className={style.statsContainer}>
            <div className={style.upperBox}>
              <div className={style.innerContent}>
                <h3 className={style.subtitle}>
                  {`${translate("match_rate")}:`}
                  <i className={style.matchRatePercent}>{Math.round(career.score)}%</i>
                </h3>
                <div className={style.matchRate}>
                  <span data-match-rate={`${career.score}%`} style={{width: `${career.score}%`}} />
                </div>
              </div>
              <div className={style.innerContent}>
                <div>
                  <h3 className={style.subtitleFull}>{`${translate("experience_level")}:`}</h3>
                  <ol className={style.experience}>
                    {[1, 2, 3, 4, 5].map((level) => (
                      <li key={level} className={career.experience_level.id >= level ? style.active : ""} />
                    ))}
                  </ol>
                </div>
              </div>
            </div>
            <div className={style.lowerBox}>
              <div className={style.innerContent}>
                <div className={style.education}>
                  <h3 className={style.subtitleFull}>{`${translate("education")}:`}</h3>
                  <h3 className={style.subtitleFull}>
                    {`${translate(`experience_level_${career.experience_level.id}`)}:`}
                  </h3>
                </div>
              </div>
              <div className={style.learnMore}>
                <button type="button" onClick={this.openModal}>Learn More</button>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
export {Career as Component};
export default withTraitify(Career);
