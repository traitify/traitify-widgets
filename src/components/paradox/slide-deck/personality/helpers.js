export function completedSlides(slides) {
  return slides.filter(({likert_response: likertResponse, response}) => (
    likertResponse || response != null
  )).map(({id, likert_response: likertResponse, response, time_taken: timeTaken}) => ({
    id,
    likert_response: likertResponse,
    response,
    time_taken: timeTaken && timeTaken >= 0 ? timeTaken : 2
  }));
}

export function isFinished(slides) {
  return slides.length > 0 && slides.length === completedSlides(slides).length;
}

export function isFullscreen() {
  const fullscreen = document.fullscreenElement
    || document.webkitFullscreenElement
    || document.mozFullScreenElement
    || document.msFullscreenElement;

  return !!fullscreen;
}

export function loadingIndex(slides) {
  return slides.findIndex((slide) => (
    !slide.loaded && !slide.likert_response && slide.response == null
  ));
}

export function slideIndex(slides) {
  return slides.findIndex((slide) => (!slide.likert_response && slide.response == null));
}

export function toggleFullscreen(options) {
  const {current, element} = options;

  if(current) {
    const exitFullscreen = document.exitFullscreen
      || document.webkitExitFullscreen
      || document.mozCancelFullScreen
      || document.msExitFullscreen;

    exitFullscreen.apply(document);
  } else {
    const requestFullscreen = element.requestFullscreen
      || element.webkitRequestFullscreen
      || element.mozRequestFullScreen
      || element.msRequestFullscreen;

    requestFullscreen.apply(element);
  }
}

export function isReady(slides) {
  if(slides.length === 0) { return false; }

  const loadedIndex = loadingIndex(slides);
  const remainingSlidesLoaded = loadedIndex === -1;
  const nextSlidesLoaded = loadedIndex > slideIndex(slides) + 2;

  return remainingSlidesLoaded || nextSlidesLoaded;
}
