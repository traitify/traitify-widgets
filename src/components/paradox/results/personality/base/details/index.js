import PropTypes from "prop-types";
import {useEffect, useState} from "react";
import {dig} from "lib/helpers/object";
import TraitifyPropTypes from "lib/helpers/prop-types";
import useDidMount from "lib/hooks/use-did-mount";
import useDidUpdate from "lib/hooks/use-did-update";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

const detailTypes = [
  {apiKey: "Complement", disableKey: "PersonalityComplements", translationKey: "complements"},
  {apiKey: "Conflict", disableKey: "PersonalityConflicts", translationKey: "conflicts"},
  {apiKey: "Environments", disableKey: "PersonalityEnvironments", translationKey: "best_work_environments"}
];

function PersonalityBaseDetails({setElement, ...props}) {
  const {assessment, getOption, personality: _personality, translate, ui} = props;
  const [activeType, setActiveType] = useState(null);
  const [types, setTypes] = useState([]);
  const personality = _personality || dig(assessment, "personality_blend")
    || dig(assessment, "personality_types", 0, "personality_type");
  const state = {activeType, personality, types};

  const disabledComponents = getOption("disabledComponents") || [];

  useDidMount(() => { ui.trigger("PersonalityBaseDetails.initialized", {props, state}); });
  useDidUpdate(() => { ui.trigger("PersonalityBaseDetails.updated", {props, state}); });
  useEffect(() => {
    if(!personality) { return; }
    if(personality.details.length === 0) { return; }

    const activeTypes = detailTypes
      .filter(({disableKey}) => !disabledComponents.includes(disableKey))
      .map((type) => {
        let data = personality.details.filter(({title}) => (title === type.apiKey))
          .map(({body}) => body);

        if(data.length === 0) {
          data = (personality[type.apiKey.toLowerCase()] || [])
            .map(({name}) => name);
        }

        return {...type, data: data.length > 1 ? data : data[0]};
      }).filter(({data}) => data);

    setTypes(activeTypes);
    setActiveType(activeTypes[0]);
  }, [personality]);

  if(disabledComponents.includes("PersonalityDetails")) { return null; }
  if(!activeType) { return null; }

  const allowHeaders = getOption("allowHeaders");
  const onChange = ({target: {value}}) => (
    setActiveType(types.find(({translationKey}) => translationKey === value))
  );

  return (
    <div className={style.container} ref={setElement}>
      {allowHeaders && <div className={style.sectionHeading}>{translate("personality_details")}</div>}
      <div className={style.tabs}>
        {types.map((type) => (
          <button
            key={type.translationKey}
            className={activeType.translationKey === type.translationKey ? style.active : ""}
            onClick={() => setActiveType(type)}
            type="button"
          >
            {translate(type.translationKey)}
          </button>
        ))}
      </div>
      <select className={style.dropdown} onChange={onChange} value={activeType.translationKey}>
        {types.map(({translationKey: key}) => (
          <option key={key} value={key}>{translate(key)}</option>
        ))}
      </select>
      <div>
        {Array.isArray(activeType.data) ? (
          <ul>
            {activeType.data.map((data) => <li key={data}>{data}</li>)}
          </ul>
        ) : (
          <div className="p">{activeType.data}</div>
        )}
      </div>
    </div>
  );
}

const personalityPropType = PropTypes.shape({
  details: PropTypes.arrayOf(
    PropTypes.shape({
      body: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired
    }).isRequired
  ).isRequired
});

PersonalityBaseDetails.defaultProps = {assessment: null, personality: null};
PersonalityBaseDetails.propTypes = {
  assessment: PropTypes.shape({
    personality_blend: personalityPropType,
    personality_types: PropTypes.arrayOf(
      PropTypes.shape({personality_type: personalityPropType.isRequired})
    )
  }),
  getOption: PropTypes.func.isRequired,
  personality: personalityPropType,
  setElement: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired
};

export {PersonalityBaseDetails as Component};
export default withTraitify(PersonalityBaseDetails);
