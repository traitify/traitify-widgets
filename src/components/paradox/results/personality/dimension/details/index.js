import PropTypes from "prop-types";
import {detailWithPerspective} from "lib/helpers";
import {findCompetency} from "lib/helpers/combine-data";
import {useDidMount, useDidUpdate} from "lib/helpers/hooks";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

function PersonalityDimensionDetails(props) {
  const {
    getOption,
    element,
    followGuide,
    guide,
    translate,
    type: {personality_type: {badge, details, id, level, name}},
    ui
  } = props;
  const state = {};

  useDidMount(() => { ui.trigger("PersonalityDimensionDetails.initialized", {props, state}); });
  useDidMount(() => { followGuide(); });
  useDidUpdate(() => { ui.trigger("PersonalityDimensionDetails.updated", {props, state}); });

  const benefits = details.filter(({title}) => (title === "Benefits")).map(({body}) => body);
  const pitfalls = details.filter(({title}) => (title === "Pitfalls")).map(({body}) => body);
  const perspective = getOption("perspective") || "firstPerson";
  const options = {base: {details}, perspective};
  const benefitsHeader = perspective === "firstPerson" ? translate("dimension_heading_for_benefits", {level, name}) : translate("potential_benefits");
  const competency = findCompetency({guide, typeID: id});
  const disabledComponents = getOption("disabledComponents") || [];
  const disablePitfalls = disabledComponents.includes("PersonalityPitfalls");

  return (
    <div className={style.container} ref={element}>
      <div className={style.header}>
        <img alt={`${name} ${translate("badge")}`} src={badge.image_medium} />
        {competency && (
          <div className={style.name}>
            {competency.name}<span className={style.divider}> | </span>
          </div>
        )}
        <div className={style.name}>{name}</div>
      </div>
      {perspective === "firstPerson" && <div className={style.heading}>{translate("dimension_heading", {level, name})}</div>}
      <div className={style.p}>{detailWithPerspective({...options, name: "short_description"})}</div>
      <div className={style.heading}>{benefitsHeader}</div>
      <div className={style.details}>
        {benefits.map((detail) => <div key={detail}>{detail}</div>)}
      </div>
      {!disablePitfalls && (
        <>
          <div className={style.heading}>{translate("room_for_growth_and_change")}</div>
          <div className={style.details}>
            {pitfalls.map((detail) => <div key={detail}>{detail}</div>)}
          </div>
        </>
      )}
    </div>
  );
}

PersonalityDimensionDetails.defaultProps = {element: null, guide: null};
PersonalityDimensionDetails.propTypes = {
  element: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({current: PropTypes.instanceOf(Element)})
  ]),
  followGuide: PropTypes.func.isRequired,
  getOption: PropTypes.func.isRequired,
  guide: PropTypes.shape({
    assessment_id: PropTypes.string.isRequired,
    competencies: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired
      }).isRequired
    ).isRequired,
    locale_key: PropTypes.string.isRequired
  }),
  translate: PropTypes.func.isRequired,
  type: PropTypes.shape({
    personality_type: PropTypes.shape({
      badge: PropTypes.shape({
        image_medium: PropTypes.string.isRequired
      }).isRequired,
      details: PropTypes.arrayOf(
        PropTypes.shape({
          body: PropTypes.string.isRequired,
          title: PropTypes.string.isRequired
        }).isRequired
      ).isRequired,
      id: PropTypes.string.isRequired,
      level: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired
    }).isRequired
  }).isRequired,
  ui: TraitifyPropTypes.ui.isRequired
};

export {PersonalityDimensionDetails as Component};
export default withTraitify(PersonalityDimensionDetails);
