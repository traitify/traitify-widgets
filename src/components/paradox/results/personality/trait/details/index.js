import PropTypes from "prop-types";
import {rgba} from "lib/helpers/color";
import TraitifyPropTypes from "lib/helpers/prop-types";
import useDidMount from "lib/hooks/use-did-mount";
import useDidUpdate from "lib/hooks/use-did-update";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

function PersonalityTraitDetails({element, ...props}) {
  const {trait: {personality_trait: trait}, ui} = props;
  const state = {};

  useDidMount(() => { ui.trigger("PersonalityTrait.initialized", {props, state}); });
  useDidUpdate(() => { ui.trigger("PersonalityTrait.updated", {props, state}); });

  const type = trait.personality_type;
  const color = `#${type.badge.color_1}`;

  return (
    <div className={style.container} ref={element} style={{background: rgba(color, 8.5)}}>
      <div className={style.bar} style={{background: color}} />
      <div className={style.content}>
        <img alt={type.name} src={type.badge.image_medium} />
        <div className={style.heading}>
          {trait.name}
          <span className={style.description}>{trait.definition}</span>
        </div>
      </div>
    </div>
  );
}

PersonalityTraitDetails.defaultProps = {element: null};
PersonalityTraitDetails.propTypes = {
  element: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({current: PropTypes.instanceOf(Element)})
  ]),
  trait: PropTypes.shape({
    personality_trait: PropTypes.shape({
      definition: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      personality_type: PropTypes.shape({
        badge: PropTypes.shape({
          color_1: PropTypes.string.isRequired,
          image_medium: PropTypes.string.isRequired
        }),
        name: PropTypes.string.isRequired
      })
    }).isRequired
  }).isRequired,
  ui: TraitifyPropTypes.ui.isRequired
};

export {PersonalityTraitDetails as Component};
export default withTraitify(PersonalityTraitDetails);
