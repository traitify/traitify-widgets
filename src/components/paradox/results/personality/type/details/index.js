import PropTypes from "prop-types";
import {useEffect, useState} from "react";
import {dig} from "lib/helpers/object";
import TraitifyPropTypes from "lib/helpers/prop-types";
import useDidMount from "lib/hooks/use-did-mount";
import useDidUpdate from "lib/hooks/use-did-update";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

const detailTypes = [
  {apiKey: "Complement", disableKey: "PersonalityComplements", translationKey: "complements"},
  {apiKey: "Conflict", disableKey: "PersonalityConflicts", translationKey: "conflicts"},
  {apiKey: "Environments", disableKey: "PersonalityEnvironments", translationKey: "best_work_environments"}
];

function PersonalityTypeDetails({setElement, ...props}) {
  const {assessment, getOption, translate, type: _personality, ui} = props;
  const [activeType, setActiveType] = useState(null);
  const [types, setTypes] = useState([]);
  const state = {activeType, types};

  const disabledComponents = getOption("disabledComponents") || [];
  const personality = _personality || dig(assessment, "personality_blend")
    || dig(assessment, "personality_types", 0, "personality_type");

  useDidMount(() => { ui.trigger("PersonalityTypeDetails.initialized", {props, state}); });
  useDidUpdate(() => { ui.trigger("PersonalityTypeDetails.updated", {props, state}); });
  useEffect(() => {
    if(!personality) { return; }
    if(personality.details.length === 0) { return; }

    const activeTypes = detailTypes
      .filter(({disableKey}) => !disabledComponents.includes(disableKey))
      .map((type) => {
        let data = personality.details.filter(({title}) => (title === type.apiKey))
          .map(({body}) => body);

        if(data.length === 0) {
          data = (personality[type.apiKey.toLowerCase()] || [])
            .map(({name}) => name);
        }

        return {...type, data: data.length > 1 ? data : data[0]};
      });

    setTypes(activeTypes);
    setActiveType(activeTypes[0]);
  }, [personality]);

  if(disabledComponents.includes("PersonalityDetails")) { return null; }
  if(!activeType) { return null; }

  const allowHeaders = getOption("allowHeaders");
  const onChange = ({target: {value}}) => (
    setActiveType(types.find(({translationKey}) => translationKey === value))
  );

  return (
    <div className={style.container} ref={setElement}>
      {allowHeaders && <div className={style.sectionHeading}>{translate("personality_details")}</div>}
      <div className={style.tabs}>
        {types.map((type) => (
          <button
            key={type.translationKey}
            className={activeType.translationKey === type.translationKey ? style.active : ""}
            onClick={() => setActiveType(type)}
            type="button"
          >
            {translate(type.translationKey)}
          </button>
        ))}
      </div>
      <select className={style.dropdown} onChange={onChange} value={activeType.translationKey}>
        {types.map(({translationKey: key}) => (
          <option key={key} value={key}>{translate(key)}</option>
        ))}
      </select>
      <div>
        {Array.isArray(activeType.data) ? (
          <ul>
            {activeType.data.map((data) => <li key={data}>{data}</li>)}
          </ul>
        ) : (
          <div className="p">{activeType.data}</div>
        )}
      </div>
    </div>
  );
}

const personalityPropType = PropTypes.shape({
  details: PropTypes.arrayOf(
    PropTypes.shape({
      body: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired
    }).isRequired
  ).isRequired
});

PersonalityTypeDetails.defaultProps = {assessment: null, type: null};
PersonalityTypeDetails.propTypes = {
  assessment: PropTypes.shape({
    personality_blend: personalityPropType,
    personality_types: PropTypes.arrayOf(
      PropTypes.shape({personality_type: personalityPropType.isRequired})
    )
  }),
  getOption: PropTypes.func.isRequired,
  setElement: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired,
  type: personalityPropType,
  ui: TraitifyPropTypes.ui.isRequired
};

export {PersonalityTypeDetails as Component};
export default withTraitify(PersonalityTypeDetails);
