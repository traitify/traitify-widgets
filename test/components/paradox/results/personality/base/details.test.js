import {Component} from "components/paradox/results/personality/base/details";
import {mutable} from "lib/helpers/object";
import ComponentHandler from "support/component-handler";
import {mockOptions, mockProps} from "support/helpers";
import assessment from "support/json/assessment/type-based.json";

jest.mock("lib/with-traitify", () => ((value) => value));

describe("Paradox.PersonalityBaseDetails", () => {
  let component;
  let options;
  let props;

  beforeEach(() => {
    options = {};
    props = {
      ...mockProps(["getOption", "setElement", "translate", "ui"]),
      assessment: mutable(assessment),
      personality: {
        details: [
          {body: "You work well with humans", title: "Complement"},
          {body: "You work poorly with aliens", title: "Conflict"}
        ],
        environments: [{name: "Home"}, {name: "Sweeeeet"}, {name: "Home?"}]
      }
    };
  });

  describe("callbacks", () => {
    it("triggers initialization", () => {
      new ComponentHandler(<Component {...props} />);

      expect(props.ui.trigger).toHaveBeenCalledWith(
        "PersonalityBaseDetails.initialized",
        expect.objectContaining({
          props: expect.any(Object),
          state: expect.any(Object)
        })
      );
    });

    it("triggers update", () => {
      component = new ComponentHandler(<Component {...props} />);
      props.ui.trigger.mockClear();
      component.updateProps();

      expect(props.ui.trigger).toHaveBeenCalledWith(
        "PersonalityBaseDetails.updated",
        expect.objectContaining({
          props: expect.any(Object),
          state: expect.any(Object)
        })
      );
    });
  });

  describe("fallbacks", () => {
    it("renders component with blend", () => {
      props.personality = null;
      component = new ComponentHandler(<Component {...props} />);

      expect(component.tree).toMatchSnapshot();
    });

    it("renders component with types", () => {
      props.assessment.personality_blend = null;
      props.personality = null;
      component = new ComponentHandler(<Component {...props} />);

      expect(component.tree).toMatchSnapshot();
    });
  });

  it("renders component", () => {
    component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders component with headers", () => {
    mockOptions(props.getOption, {...options, allowHeaders: true});
    component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders component with no environments", () => {
    props.personality.environments = null;
    component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders new type from clicking button", () => {
    component = new ComponentHandler(<Component {...props} />);
    component.act(() => component.instance.findAllByType("button")[1].props.onClick());

    expect(component.tree).toMatchSnapshot();
  });

  it("renders new type from selecting option", () => {
    component = new ComponentHandler(<Component {...props} />);
    const select = component.instance.findByType("select");
    const option = component.instance.findAllByType("option")[2];
    component.act(() => select.props.onChange({target: option.props}));

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if disabled", () => {
    mockOptions(props.getOption, {...options, disabledComponents: ["PersonalityDetails"]});
    component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if no details", () => {
    props.personality.details = [];
    component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if results not ready", () => {
    props.assessment = null;
    props.personality = null;
    component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });
});
