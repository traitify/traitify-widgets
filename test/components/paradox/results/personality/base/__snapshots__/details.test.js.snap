// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paradox.PersonalityBaseDetails fallbacks renders component with blend 1`] = `
<div
  className="container"
>
  <div
    className="tabs"
  >
    <button
      className="active"
      onClick={[Function]}
      type="button"
    >
      complements
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      conflicts
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      best_work_environments
    </button>
  </div>
  <select
    className="dropdown"
    onChange={[Function]}
    value="complements"
  >
    <option
      value="complements"
    >
      complements
    </option>
    <option
      value="conflicts"
    >
      conflicts
    </option>
    <option
      value="best_work_environments"
    >
      best_work_environments
    </option>
  </select>
  <div>
    <div
      className="p"
    >
      You often find yourself working with other Naturalists as you try to work outdoors as much as possible and find your like-minded colleagues alongside you.  You also enjoy working with other Analyzers whose shared cerebral interests find you having many things in common. You appreciate your Planner colleagues as well.  They take care of managing the details so you can think through the big picture of a project and then get started on it right away.

    </div>
  </div>
</div>
`;

exports[`Paradox.PersonalityBaseDetails fallbacks renders component with types 1`] = `
<div
  className="container"
>
  <div
    className="tabs"
  >
    <button
      className="active"
      onClick={[Function]}
      type="button"
    >
      complements
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      conflicts
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      best_work_environments
    </button>
  </div>
  <select
    className="dropdown"
    onChange={[Function]}
    value="complements"
  >
    <option
      value="complements"
    >
      complements
    </option>
    <option
      value="conflicts"
    >
      conflicts
    </option>
    <option
      value="best_work_environments"
    >
      best_work_environments
    </option>
  </select>
  <div>
    <div
      className="p"
    >
      You often find yourself working.
    </div>
  </div>
</div>
`;

exports[`Paradox.PersonalityBaseDetails renders component 1`] = `
<div
  className="container"
>
  <div
    className="tabs"
  >
    <button
      className="active"
      onClick={[Function]}
      type="button"
    >
      complements
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      conflicts
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      best_work_environments
    </button>
  </div>
  <select
    className="dropdown"
    onChange={[Function]}
    value="complements"
  >
    <option
      value="complements"
    >
      complements
    </option>
    <option
      value="conflicts"
    >
      conflicts
    </option>
    <option
      value="best_work_environments"
    >
      best_work_environments
    </option>
  </select>
  <div>
    <div
      className="p"
    >
      You work well with humans
    </div>
  </div>
</div>
`;

exports[`Paradox.PersonalityBaseDetails renders component with headers 1`] = `
<div
  className="container"
>
  <div
    className="sectionHeading"
  >
    personality_details
  </div>
  <div
    className="tabs"
  >
    <button
      className="active"
      onClick={[Function]}
      type="button"
    >
      complements
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      conflicts
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      best_work_environments
    </button>
  </div>
  <select
    className="dropdown"
    onChange={[Function]}
    value="complements"
  >
    <option
      value="complements"
    >
      complements
    </option>
    <option
      value="conflicts"
    >
      conflicts
    </option>
    <option
      value="best_work_environments"
    >
      best_work_environments
    </option>
  </select>
  <div>
    <div
      className="p"
    >
      You work well with humans
    </div>
  </div>
</div>
`;

exports[`Paradox.PersonalityBaseDetails renders component with no environments 1`] = `
<div
  className="container"
>
  <div
    className="tabs"
  >
    <button
      className="active"
      onClick={[Function]}
      type="button"
    >
      complements
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      conflicts
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      best_work_environments
    </button>
  </div>
  <select
    className="dropdown"
    onChange={[Function]}
    value="complements"
  >
    <option
      value="complements"
    >
      complements
    </option>
    <option
      value="conflicts"
    >
      conflicts
    </option>
    <option
      value="best_work_environments"
    >
      best_work_environments
    </option>
  </select>
  <div>
    <div
      className="p"
    >
      You work well with humans
    </div>
  </div>
</div>
`;

exports[`Paradox.PersonalityBaseDetails renders new type from clicking button 1`] = `
<div
  className="container"
>
  <div
    className="tabs"
  >
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      complements
    </button>
    <button
      className="active"
      onClick={[Function]}
      type="button"
    >
      conflicts
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      best_work_environments
    </button>
  </div>
  <select
    className="dropdown"
    onChange={[Function]}
    value="conflicts"
  >
    <option
      value="complements"
    >
      complements
    </option>
    <option
      value="conflicts"
    >
      conflicts
    </option>
    <option
      value="best_work_environments"
    >
      best_work_environments
    </option>
  </select>
  <div>
    <div
      className="p"
    >
      You work poorly with aliens
    </div>
  </div>
</div>
`;

exports[`Paradox.PersonalityBaseDetails renders new type from selecting option 1`] = `
<div
  className="container"
>
  <div
    className="tabs"
  >
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      complements
    </button>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      conflicts
    </button>
    <button
      className="active"
      onClick={[Function]}
      type="button"
    >
      best_work_environments
    </button>
  </div>
  <select
    className="dropdown"
    onChange={[Function]}
    value="best_work_environments"
  >
    <option
      value="complements"
    >
      complements
    </option>
    <option
      value="conflicts"
    >
      conflicts
    </option>
    <option
      value="best_work_environments"
    >
      best_work_environments
    </option>
  </select>
  <div>
    <ul>
      <li>
        Home
      </li>
      <li>
        Sweeeeet
      </li>
      <li>
        Home?
      </li>
    </ul>
  </div>
</div>
`;

exports[`Paradox.PersonalityBaseDetails renders nothing if disabled 1`] = `null`;

exports[`Paradox.PersonalityBaseDetails renders nothing if no details 1`] = `null`;

exports[`Paradox.PersonalityBaseDetails renders nothing if results not ready 1`] = `null`;
